TopMovies/API/GraphQL/NetworkManager.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  NetworkManager.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 01.04.2021.
+//
+
+import Foundation
+
+struct QueryJSON: Encodable {
+  let query: String
+}
+
+class NetworkManager {
+  func requestAsync(with queryJSON: QueryJSON, completion: @escaping (Result<Data, Error>) -> Void) {
+    let session = URLSession(configuration: .default)
+    var request = URLRequest(url: URL(string: "https://tmdb.apps.quintero.io/")!)
```

Comment:
Why not using `Moya` interface for that networking layer too? 
You could iterate by enum's cases right the same way as you described for the REST API. 

Here, we have 1 method that: configures request and session, provides base url, http method, headers, body, performs data encoding, performs the request, and so on.